lib/graphql/language/nodes.rb

```diff
@@ -212,6 +216,14 @@ def initialize_node(name: nil, arguments: [], directives: [], selections: [], **
           @directives = directives
           @selections = selections
         end
+
+        def eql?(other)
+          hash == other.hash
+        end
```

Comment:
I don't think this is quite right.  Two objects could hash to the same value, but not be equal (that is a hash collision).